torch/_inductor/comms.py

```diff
@@ -342,6 +342,148 @@ def reorder_compute_and_comm_for_overlap(
     return order
 
 
+def remove_fsdp2_unsharded_param_graph_input_usage(graph: torch.fx.Graph):
+    """
+    This FX graph pass replaces uses of FSDP2 unsharded params with their corresponding
+    graph intermediates that were fsdp.copy_ into the unsharded params in the original graph.
+
+    NOTE: Can only apply this pass to FSDP2 unsharded params that are resharded (i.e. resized to 0)
+    at the end of graph.
+    """
+    node_list = list(graph.nodes)
+
+    # Find all graph inputs and their resize counts
+    graph_input_to_resized_to_full_node_idxes = defaultdict(list)
+    graph_input_to_resized_to_0_node_idxes = defaultdict(list)
+    for idx, node in enumerate(node_list):
+        if (
+            node.op == "call_function"
+            and node.target == torch.ops.inductor.resize_storage_bytes_.default
+            and node.args[0].op == "placeholder"
+        ):
+            graph_input = node.args[0]
+            new_size = node.args[1]
+            if new_size > 0:
+                graph_input_to_resized_to_full_node_idxes[graph_input].append(idx)
```

Comment:
hmmm so.... maybe you've told me this before and I just forgot, but why is it possible to end up with a graph that has `resize_storage_bytes_()` resize the same placeholder up or down multiple times in a single graph?

Does this happen if you have a parameter that is shared across multiple wrapped FSDP layers? (so the usage pattern is something like `resize up -> resize down -> resize up -> resize down` in a single graph)